### PR TITLE
build android sdk with android-10

### PR DESCRIPTION
**Pull request by sunchunqiang**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -188,6 +188,23 @@ jobs:
...
+    working_directory: ~/code
+    docker:
+      - image: circleci/android:api-27-alpha
+    environment:
+      JVM_OPTS: -Xmx3200m
+    steps:
+      - checkout
+      - generate_gradle_wrapper
+      - run:
+          name: downloand android-10
```
Comment (minbi) [inline .circleci/config.yml]: download

File: .circleci/config.yml [hunk 2]
```diff
@@ -188,6 +188,23 @@ jobs:
...
+          command:
+            sudo yes | sdkmanager "platforms;android-23"
```
Comment (minbi) [inline .circleci/config.yml]: android-10

File: .circleci/config.yml [hunk 3]
```diff
@@ -188,6 +188,23 @@ jobs:
...
+      - run:
+          name: build the whole project
+          command: |
+            bash gradlew build -x test
```
Comment (minbi) [inline .circleci/config.yml]: this needs to build just core

File: .circleci/config.yml [hunk 4]
```diff
@@ -603,6 +620,7 @@ workflows:
   build_test:
     jobs:
       - build
+      - build_api10
```
Comment (minbi) [inline .circleci/config.yml]: this should be added to release_sdk as well

File: CircleciScripts/replace_android10.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from utils import replacefiles
+import demjson
+import sys
+
+root = sys.argv[1]
+
+replaces = [
+    {
+        "match" : 'android-23',
```
Comment (minbi) [inline CircleciScripts/replace_android10.py]: there needs to be a way to remove the relevant code that uses api 23 and 18. This code is safe-guarded during runtime, but this ensure the rest of the code adheres to api level 10. Check with @kvasukib  which code parts should be removed

File: .circleci/config.yml (hunk 1 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: addressed

File: .circleci/config.yml (hunk 3 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: addressed

File: CircleciScripts/replace_android10.py (hunk 5 above)
Comment (sunchunqiang) [reply CircleciScripts/replace_android10.py]: addressed

File: .circleci/config.yml [hunk 6]
```diff
@@ -1216,6 +1256,7 @@ workflows:
       - release_s3:
           requires:
             - build
+            - build_api10
```
Comment (minbi) [inline .circleci/config.yml]: this should be required by release_maven, not s3

File: CircleciScripts/replace_android10.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+replacefiles(root, replaces)
+
+
+
+
+AWSKeyValueStoreFile = os.path.join(root,"aws-android-sdk-core/src/main/java/com/amazonaws/internal/keyvaluestore/AWSKeyValueStore.java")
+newcontent = ""
+with open(AWSKeyValueStoreFile, 'r') as myfile:
+    content = myfile.read() 
+    pattern = r'if\s*\(apiLevel >= ANDROID_API_LEVEL_23\)\s*\{[^\}]+\}\s*else if\s*\(apiLevel >= ANDROID_API_LEVEL_18\)\s*\{[^\}]+\}'
```
Comment (minbi) [inline CircleciScripts/replace_android10.py]: @kvasukib had added some annotations something along the lines of `@api23start` and `@api23end` would that be a better way to replace?

**Pull request merged**